public/components/datasources/components/manage/data_connection.tsx

```diff
@@ -294,6 +344,55 @@ export const DataConnection = (props: any) => {
     );
   };
 
+  useEffect(() => {
+    // Loads databases cache when cache is empty
+    if (datasourceDetails.name) {
+      const cache = CatalogCacheManager.getDataSourceCache().dataSources.find(
+        (datasource) => datasource.name === datasourceDetails.name
+      );
+      if (cache?.status !== 'Updated') {
```

Comment:
Not sure if this is a potential issue or expected behavior: if cache is null this check will also be evaluated as truthy, would this also be part of the handling here or no?